### DISCUSSION TITLE
(side) download fails on hos.com

### DISCUSSION THREAD

**Discussion by Naked-Driver** [aclap-dev/video-downloadhelper#209] [Help and Q&A]
### Your question

when i try to d/l the program of the week, i get this message:
[begin pasted text] SideDownload error:

[mp3 @ 0x44bcd80] Invalid audio stream. Exactly one MP3 audio stream is required.
[out#0/mp3 @ 0x44a1800] Could not write header (incorrect codec parameters ?): Invalid argument
[aost#0:0/copy @ 0x44bb500] Error initializing output stream: [end pasted text]
for some reason, it lists three mp3 streams!
why not go back to just the mp4, which worked, and i can convert it myself?
thanks!

### Addon & CoApp version and details

version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "/opt/vdhcoapp-2.0.10/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/naked_driver",
  "converterBinary": "/opt/vdhcoapp-2.0.10/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by jcfrog** [comment]
Hi,
Do you have a public URL to reproduce this issue?

**Comment by paulrouget** [comment]
I can reproduce (https://v4.hos.com/home).

I think I know what's going on, i'll look into it.

**Reply by GrampaWildWilly** [reply]
I got the URL of the HLS master manifest that shows up in the Network Monitor & pushed that through ffprobe.  It told me 403 Forbidden.  If you have a way around that, I'd love to learn what it is.

**Comment by paulrouget** [comment]
I found the issue. Will be fixed in 8.2.1.4.

**Reply by GrampaWildWilly** [reply]
I just installed 8.2.1.5a1 beta & that hos.com link still just shows me a bunch of mp2t variants.  I asked above about the 403 Forbidden I got on the HLS master manifest.  Could the failure be related to that?

**Reply by paulrouget** [reply]
MP2T variants? You don't see this?

<img width="497" alt="Screenshot 2024-02-29 at 16 58 03" src="https://github.com/aclap-dev/video-downloadhelper/assets/373579/cbbda0ed-d92c-41b2-ad49-ca44b006f11a">

**Reply by GrampaWildWilly** [reply]
Well, I have some things turned off so I get more variants than the norm, I think.  But now that I've gone to the page again just now, I am seeing mp3 variants that I wasn't before.  It looks like this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/5b927353-792e-4e00-9c8b-62412500fe81)

I swear those mp3 variants were most definitely NOT there when I went there before.  I guess the page reload made a difference.  But I reloaded the page before & it didn't help.  I don't know, man . . .

I downloaded the first variant on the menu.  (It's showing in my image as the blue dot at the bottom of the VDH menu.)  The speed was only in the 210,000 bytes per second range according to Resource Monitor.  Curiously, VDH was reporting about 125K bytes per second.  But I noted in the Resource Monitor that there were 2 instances of ffmpeg running:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/0183777e-5c4e-4e7e-90d5-c58928ee855b)

In any case, it did eventually complete downloading:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/df0371ef-8430-4f08-9f0d-730902f395df)

Played fine in VLC, although I only sampled it.

I must say it's a bit confusing that you & I both have 5 variants marked simply mp3 without any other information to distinguish them.  How do you correlate the VDH menu with the contents of the page?  It looks like there's only one audio item on the page.  So why the multiple variants?  Are they presenting different bit rates of this item?  I would answer that question myself but ffprobe on the HLS master manifest once again gave me 403 Forbidden.  What's up with that?  How am I getting this error but you are able to download this stuff anyway?  In any case, here's some circumstantial evidence that there are different bit rate variants being offered:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/1ca5a110-c1cf-4c71-b0a6-d781ca912120)

There is a 256k variant & a 128k variant, but my results above say it's a 124k file.  Is that a sorting error in VDH?  Or are the bit rates not actually what you would think based on the URLs in the manifest?  Since I can't get an ffprobe of the thing, I can only guess.
